components/script/dom/htmlbaseelement.rs

```diff
@@ -63,8 +63,8 @@ impl HTMLBaseElement {
 
     /// Update the cached base element in response to binding or unbinding from
     /// a tree.
-    pub fn bind_unbind(&self, tree_in_doc: bool) {
-        if !tree_in_doc {
+    pub fn bind_unbind(&self, tree_connected: bool) {
+        if !tree_connected {
```

Comment:
This is wrong, `<base>` in a shadow tree should do nothing. Can we have a `BindContext` just like we have `UnbindContext` that has both `tree_in_doc` and `tree_connected`?

Again, should be doable to add a test for this.